app/services/course/user_invitation_service.rb

```diff
@@ -63,13 +65,25 @@ def resend_invitation(invitations)
   #     Array<(Array<Course::UserInvitation>,
   #     Array<Course::UserInvitation>,
   #     Array<CourseUser>,
-  #     Array<CourseUser>)>
+  #     Array<CourseUser>)>,
+  #     Array<Hash>,
   #   ]
   #   A tuple containing the users newly invited, already invited,
-  #     newly registered and already registered respectively.
+  #     newly registered and already registered, and duplicate users respectively.
   # @raise [CSV::MalformedCSVError] When the file provided is invalid.
   def invite_users(users)
-    users = parse_invitations(users)
-    process_invitations(users)
+    users, duplicate_users = parse_invitations(users)
+
+    # Process the duplicate users once instead of totally ignoring them.
+    unique_duplicates = duplicate_users.uniq { |user| user[:email] }
+    users += unique_duplicates
+    # Remove one of each duplicate user from the duplicate_users array so the
+    # count in the warning message will be correct.
+    unique_duplicates.each do |user|
+      index = duplicate_users.index(user)
+      duplicate_users.delete_at(index) if index
+    end
```

Comment:
This logic is unnecessarily awkward. Instead, in the case of duplicates you could let parse_invitations put the first duplicate instance into the success pile and the rest into the skipped pile.